aiida/cmdline/commands/cmd_storage.py

```diff
@@ -85,41 +85,100 @@ def storage_integrity():
 
     # Group
     query_group = QueryBuilder().append(Group, project=['type_string'])
-    data['Groups'] = {'count': query_group.count()}
+    dbdata['Groups'] = {'count': query_group.count()}
     if statistics:
-        data['Groups']['type_strings'] = query_group.distinct().all(flat=True)
+        dbdata['Groups']['type_strings'] = query_group.distinct().all(flat=True)
 
     # Comment
     count = QueryBuilder().append(Comment).count()
-    data['Comments'] = {'count': count}
+    dbdata['Comments'] = {'count': count}
 
     # Log
     count = QueryBuilder().append(Log).count()
-    data['Logs'] = {'count': count}
+    dbdata['Logs'] = {'count': count}
+
+    repodata = repository_info(statistics=statistics)
+
+    echo.echo_dictionary({'database': dbdata, 'repository': repodata}, sort_keys=False, fmt='yaml')
+
+
+@verdi_storage.command('maintain')
+@click.option(
+    '--full',
+    is_flag=True,
+    help='Performs maintenance tasks that are safe to run while other AiiDA instances are working with the profile.'
+)
+@click.option(
+    '--dry-run',
+    is_flag=True,
+    help='Performs maintenance tasks that are safe to run while other AiiDA instances are working with the profile.'
+)
+def storage_maintain(full, dry_run):
+    """Performs maintenance tasks on the repository."""
+    from aiida.repository.control import get_repository_report, repository_maintain
+
+    if dry_run and full:
+        echo.echo_critical('You cannot request both `--dry-run` and `--full` at the same time.')
+
+    if dry_run:
+        maintainance_report = get_repository_report()['user_info']
+        click.echo('Repository:')
+        click.echo(maintainance_report)
+        return
+
+    if full:
+
+        click.echo(
+            '\nIn order to safely perform the full maintenance operations on the internal storage, no other '
+            'process should be using the AiiDA profile being maintained. '
+            'This includes daemon workers, verdi shells, scripts with the profile loaded, etc). '
+            'Please make sure there is nothing like this currently running and that none is started until '
+            'these procedures conclude. '
+            'For performing maintanance operations that are safe to run while actively using AiiDA, just run '
+            '`verdi storage maintain`, without the `--full` flag.\n'
+        )
+
+    else:
+
+        click.echo(
+            '\nThis command will perform all maintenance operations on the internal storage that can be safely '
+            'executed while still running AiiDA. '
+            'However, not all operations that are required to fully optimize disk usage and future performance '
+            'can be done in this way. '
+            'Whenever you find the time or opportunity, please consider running `verdi repository maintenance '
+            '--full` for a more complete optimization.\n'
+        )
+
+    if not click.confirm('Are you sure you want continue in this mode?'):
+        return
```

Comment:
Since the difference between maintenance types is just the impossibility to use the profile, the partial maintenance can still be a complex lengthy task that the user might not want to start by error because he forgot to add a keyword. Also if at some point we add some "verbosity" option that outputs a lot of information while processing, it gives time for the user to actually read the message before it gets buried. I don't know, I figured it wouldn't hurt since it is not a command you are calling frequently anyways and when you do call it you want to make sure you are using the right one. If you are still against it I can take it out, I don't think it is critical.

Reply:
> I don't know, I figured it wouldn't hurt since it is not a command you are calling frequently anyways and when you do call it you want to make sure you are using the right one.

I was under the impression that the `verdi storage maintain` should be expected to be run often, exactly because it can be done while running AiiDA. So if that is true, it would be a bit annoying if there is a prompt everytime, especially if there is no force option. But also for me it is not critical. If someone else has an opinion either way, that would be fine for me.